app/src/main/java/com/example/groceriesmanager/Adapters/RecipeSearchAdapter.java

```diff
@@ -140,14 +140,18 @@ public void bind(Recipe recipe) {
                     recipeIngredientStr = recipeIngredientStr + name + "\r\n";
                 }
 
-                // check if ingredient with same name in pantry
-                for (FoodItem pantryItem: pantryList){
-                    // todo: when you get lemmatizer working, use lemmatizer instead of substring match
-                    if (Objects.equals(name.toLowerCase(), pantryItem.getName().toLowerCase())||name.toLowerCase().contains(pantryItem.getName().toLowerCase())){
-                        gottenIngredients ++;
-                        addAllList.remove(ingredient); // if already in pantry, remove ingredient from list
+                // todo: find a better way of implementing pantry list checks that does not require the user to open the pantry list first before oepning this fragment. this will help eliminate this null check.
```

Comment:
You could potentially have an async function that queries for the pantry list and registers an onComplete listener.  This would allow you to put up a spinner while this is happening ( assuming the pantry list needs to be populated or is large ) and will allow you know when the pantryList is valid or not.  I would still recommend doing a null check just for a sanity check it's always good but this could prevent you from having to open the pantry first.  Just a thought